socketcan_interface/CMakeLists.txt

```diff
@@ -1,151 +1,132 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(socketcan_interface)
 
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
-find_package(catkin REQUIRED
-  COMPONENTS
-    class_loader
-)
-
-find_package(console_bridge REQUIRED)
-
+find_package(ament_cmake REQUIRED)
+find_package(pluginlib REQUIRED)
 find_package(Boost REQUIRED
   COMPONENTS
     chrono
     system
     thread
 )
-
 find_package(Threads REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}_string
-  CATKIN_DEPENDS
-  DEPENDS
-    Boost
-    console_bridge
-)
-
-include_directories(
-  include
+include_directories(include
+  ${pluginlib_INCLUDE_DIRS}
   ${Boost_INCLUDE_DIRS}
-  ${catkin_INCLUDE_DIRS}
-  ${console_bridge_INCLUDE_DIRS}
 )
 
 # ${PROJECT_NAME}_string
-add_library(${PROJECT_NAME}_string
-    src/string.cpp
+add_library(${PROJECT_NAME}_string SHARED
+  src/string.cpp
 )
 
 # socketcan_dump
-add_executable(socketcan_dump
-  src/candump.cpp
-)
-
+add_executable(socketcan_dump src/candump.cpp)
 target_link_libraries(socketcan_dump
-   ${PROJECT_NAME}_string
-   ${console_bridge_LIBRARIES}
-   ${catkin_LIBRARIES}
-   ${Boost_LIBRARIES}
-   ${CMAKE_THREAD_LIBS_INIT}
+  ${PROJECT_NAME}_string
+  ${pluginlib_LIBRARIES}
+  ${Boost_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
 )
 
 # socketcan_bcm
-add_executable(socketcan_bcm
-  src/canbcm.cpp
-)
-
+add_executable(socketcan_bcm src/canbcm.cpp)
 target_link_libraries(socketcan_bcm
-   ${PROJECT_NAME}_string
-   ${console_bridge_LIBRARIES}
-   ${catkin_LIBRARIES}
-   ${Boost_LIBRARIES}
+  ${PROJECT_NAME}_string
+  ${Boost_LIBRARIES}
 )
 
 # ${PROJECT_NAME}_plugin
-add_library(${PROJECT_NAME}_plugin
+add_library(${PROJECT_NAME}_plugin SHARED
   src/${PROJECT_NAME}_plugin.cpp
 )
 target_link_libraries(${PROJECT_NAME}_plugin
-  ${console_bridge_LIBRARIES}
-  ${catkin_LIBRARIES}
+  ${pluginlib_LIBRARIES}
   ${Boost_LIBRARIES}
 )
 
+pluginlib_export_plugin_description_file(${PROJECT_NAME} socketcan_interface_plugin.xml)
+
+ament_export_dependencies(
+  ament_cmake
```

Comment:
why does `ament_cmake` need to be exported?

Reply:
No. Removed.